pkg/kubelet/volume_host.go

```diff
@@ -183,6 +185,26 @@ func (kvh *kubeletVolumeHost) WaitForCacheSync() error {
 	return nil
 }
 
+// WaitForNodeRegistrationCompleted is a helper function that waits for node is in registrationCompleted state
+func (kvh *kubeletVolumeHost) WaitForNodeRegistrationCompleted(ctx context.Context) error {
+	err := wait.PollUntilContextCancel(ctx, time.Second, true, kvh.verifyNodeRegistrationCompleted())
+	if err != nil {
+		// In theory this is unreachable, but just in case:
+		return fmt.Errorf("faled to WaitForNodeRegistrationCompleted: %v", err)
+	}
+	return nil
+}
+
+func (kvh *kubeletVolumeHost) verifyNodeRegistrationCompleted() wait.ConditionWithContextFunc {
+	return func(_ context.Context) (done bool, err error) {
+		if kvh.kubelet.registrationCompleted {
```

Comment:
The comment says this should only be accessed by `registerWithApiserver`. We may need to make it thread safe: https://github.com/kubernetes/kubernetes/blob/5cf72fbfae29c79d93cfb191bfbaf7392f28f093/pkg/kubelet/kubelet.go#L1069